### PR TITLE
Setup: GCP workload identity configuration

### PR DESCRIPTION
**Pull request by biancaburtoiu**
Document steps to set up Workload Identity federation for GCP clusters. 

image
image
image

### CODE REVIEW

File: gcp-out-of-cluster.md [hunk 1]
```diff
@@ -37,35 +37,55 @@ gcloud projects add-iam-policy-binding $PROJECT_ID --member serviceAccount:compu
...
+In Kubecost, select _Settings_ from the left navigation, and under External Cloud Cost Configuration (GCP), select _Update_, then follow the remaining instructions found at the "Add Key" link:
 
 ![GCP out-of-cluster key entry](https://raw.githubusercontent.com/kubecost/docs/main/images/gcp-out-of-cluster-config-wo-shell.png)
 
+<a name="bq-name"></a>**BigQuery dataset** requires a BigQuery dataset prefix (e.g. billing_data) in addition to the BigQuery table name. A full example is `billing_data.gcp_billing_export_v1_018AIF_74KD1D_534A2`
 
-<a name="bq-name"></a>**BigQuery dataset** requires a BigQuery dataset prefix (e.g. billing_data) in addition to the BigQuery table name. A full example is `billing_data.gcp_billing_export_v1_018AIF_74KD1D_534A2`.
+<a name="bq-name"></a>**Service key** If you've created a service account key, copy the contents of the `compute-viewer-kubecost-key.json` file and paste them here (be careful with this!). If you've connected using Workload Identity federation, you should leave this box empty.
 
 ### Configuring using values.yaml (Recommended)
```
Comment (Sean-Holcomb) [inline gcp-out-of-cluster.md:74]: We might want to update WI to be recommended

File: gcp-out-of-cluster.md [hunk 2]
```diff
@@ -37,35 +37,50 @@ gcloud projects add-iam-policy-binding $PROJECT_ID --member serviceAccount:compu
...
+It is recommended to provide the GCP details in the [values file](https://github.com/kubecost/cost-analyzer-helm-chart/blob/c10e9475b51612d36da8f04618174a98cc62f8fd/cost-analyzer/values.yaml#L572-L574) to ensure they are retained during an upgrade or redeploy.
 
-Then create a secret for the GCP service account key:
+* Set `.Values.kubecostProductConfigs.projectID = <GCP Project ID that contains the BigQuery Export>`
+* Set `.Values.kubecostProductConfigs.bigQueryBillingDataDataset = <DATASET.TABLE_NAME that contains the billing export>`
+
+If you've connected using Workload Identity federation:
+* Set `.Values.nodeSelector = iam.gke.io/gke-metadata-server-enabled: "true"` to update the Kubecost deployment to run on nodes that use Workload Identity
+* Set `.Values.serviceAccount.annotations = iam.gke.io/gcp-service-account: compute-viewer-kubecost@$PROJECT_ID.iam.gserviceaccount.com` where $PROJECT_ID defined in the `gcloud` commands above
```
Comment (biancaburtoiu) [inline gcp-out-of-cluster.md:77]: @Sean-Holcomb I've also added a `nodeSelector` constraint and moved the deployment `annotation` to the helm params

**Review by bstuder99 — APPROVED** [review]
LGTM

File: gcp-out-of-cluster.md (hunk 2 above)
Comment (Sean-Holcomb) [reply gcp-out-of-cluster.md:79]: very good

**Review by Sean-Holcomb — APPROVED** [review]

**Comment by srpomeroy** [comment]
Waiting for 1.98 to merge.

**Pull request merged**